recipes/android-ndk/all/conanfile.py

```diff
@@ -294,12 +298,25 @@ def package_info(self):
         self.conf_info.define("tools.android:ndk_path", os.path.join(self.package_folder, "bin"))
 
         compiler_executables = {
-            "c": self._define_tool_var("CC", "clang"),
-            "cpp": self._define_tool_var("CXX", "clang++"),
+            "c": self._define_tool_var_naked("CC", "clang"),
+            "cpp": self._define_tool_var_naked("CXX", "clang++"),
         }
+
+        # https://github.com/android/ndk/issues/1856
+        compiler_flags = [f"--target={self._clang_target}"]
+
+        # For x86 targets prior to Android Nougat (API 24), -mstackrealign is needed to properly align stacks for global constructors. See https://github.com/android/ndk/issues/635.
+        # https://android.googlesource.com/platform/ndk/+/refs/heads/ndk-release-r21/docs/BuildSystemMaintainers.md#additional-required-arguments
+        if self.settings_target.arch == "x86" and int(str(self.settings_target.os.api_level)) < 24:
+            compiler_flags.append("-mstackrealign")
```

Comment:
I am using NDK version 23.1.7779620 (r23b). The change you linked appears to be a fix for the wrapper scripts. Since this PR aims to stop using the wrapper scripts, the flags need to be passed manually.